scripts/update_institution_sso_email_domain.py

```diff
@@ -0,0 +1,117 @@
+import argparse
+from enum import IntEnum
+import logging
+
+from website.app import setup_django
+setup_django()
+
+from framework import sentry
+from framework.auth import get_user
+
+from osf.models import Institution, OSFUser
+
+logger = logging.getLogger(__name__)
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-i', '--id', required=True,
+                    help='the institution of which affiliated users\' eligible email is to be updated')
+parser.add_argument('-s', '--src', required=True, help='the email domain that are eligible for update')
+parser.add_argument('-d', '--dst', required=True, help='the email domain that is to be added')
+
+
+class UpdateResult(IntEnum):
+    """Defines 4 states of the email update outcome.
+    """
+    SUCCEEDED = 0  # The email has successfully been added
+    SKIPPED = 1  # The email was added (to the user) before the script is run, or the eligible email is not found
+    FAILED = 2  # The email failed to be added since it belongs to another account
+    ERRORED = 3  # The email failed to be added due to unexpected exceptions
+
+
+def main():
+    """Update the emails of users from the affiliated institution (when eligible). Requires three arguments:
+    `institution_id`, `src_domain` and `dst_domain`. The domains do not go through email validation or spam check
+    since the script can only be run by trusted and privileged admins.
+    """
+
+    args = parser.parse_args()
+    institution_id = args.id
+    src_domain = args.src
+    dst_domain = args.dst
+
+    # Verify the institution
+    institution = Institution.load(institution_id)
+    if not institution:
+        message = 'Error: invalid institution ID [{}]'.format(institution_id)
+        logger.error(message)
+        sentry.log_message(message)
+        return
+
+    # Find all users affiliated with the given institution
+    affiliated_users = OSFUser.objects.filter(affiliated_institutions___id=institution_id)
```

Comment:
Minor: Consider adding an `.exclude` subquery here to filter out users that have already been updated? Not strictly necessary, as that case results in `SKIPPED`, but may be viable way to optimize if this takes too long.

Reply:
Will add the filter. I thought the "skipped" list would be useful but probably not since it's not something we will have to manually deal later (unlike the failed ones).

Reply:
hmm, this seems not doable. It is hard to tell at this time whether a user has been updated or not since we can't just rely on whether user has an `@osf.io` email. e.g. if user has `email1@email.osf.io`, `email2@email.osf.io` and `email2@osf.io`; or if user has `email1@email.osf.io` and only `email2@osf.io`.

Reply:
It's technically possible, but may not be supported by the DjangORM, and might require `RawSQL`. Not worth the effort unless this script ends up taking significantly longer than I expect it to.